src/shims/unix/linux/sync.rs

```diff
@@ -193,14 +193,7 @@ pub fn futex<'tcx>(
                     this.register_timeout_callback(
                         thread,
                         timeout_time,
-                        Box::new(move |this| {
-                            this.unblock_thread(thread);
-                            this.futex_remove_waiter(addr_usize, thread);
-                            let etimedout = this.eval_libc("ETIMEDOUT")?;
-                            this.set_last_error(etimedout)?;
-                            this.write_scalar(Scalar::from_machine_isize(-1, this), &dest)?;
-                            Ok(())
-                        }),
+                        Box::new(Callback { thread, addr_usize, dest }),
```

Comment:
Can you define this struct inline in this function? I think we can and should keep the code together in one place, like it was before with the closure.